geomstats/geometry/pullback_metric.py

```diff
@@ -273,64 +165,30 @@ def mean_curvature_vector(self, base_point):
         mean_curvature_vector : array-like, shape=[..., embedding_dim]
             Mean curvature vector.
         """
-        base_point = gs.to_ndarray(base_point, to_ndim=2)
-
-        mean_curvature = []
-        for point in base_point:
-            second_fund_form = self.second_fundamental_form(point)
-            cometric = self.cometric_matrix(point)
-            mean_curvature.append(gs.einsum("ij,aij->a", cometric, second_fund_form))
-
-        return gs.stack(mean_curvature, axis=0)
+        second_fund_form = self.second_fundamental_form(base_point)
+        cometric = self.cometric_matrix(base_point)
+        return gs.einsum("...ij,...aij->...a", cometric, second_fund_form)
 
 
 class PullbackDiffeoMetric(RiemannianMetric, abc.ABC):
-    """
-    Pullback metric via a diffeomorphism.
-
-    Parameters
-    ----------
-    dim : int
-        Dimension.
-    shape : tuple of int
-        Shape of one element of the underlying manifold.
-        Optional, default : None.
-    """
+    """Pullback metric via a diffeomorphism."""
 
-    def __init__(self, dim, shape=None):
-        super().__init__(dim=dim, shape=shape)
+    def __init__(self, space, signature=None):
```

Comment:
Why does this metric have an init? It seems that the logic is different here?

To match the logic of the other pullback metric, it should be a metric for which the space has to be defined by an immersion that is a diffeomorphism.

And we wouldn't need embedding space bcs that would come from the immersed set.

Reply:
There's indeed a logic difference here. I think this is another part where we are still experimenting and haven't converged to an architectural solution.

This is my interpretation.

An `immersion` defines a manifold (that's why we came up with the notion of `ImmersedSet`). e.g. the point representation we use for computations is dependent on the choice of the immersion. A `PullbackMetric` is defined on a manifold defined by an immersion.

On the other hand, a `PullbackDiffeoMetric` defines a **metric** through a diffeomorphism. So we start with two manifolds $M$ and $N$ and then we "assign" a diffeormorphism ($\phi: M \to N$) between them. $M$ can exist without the diffeormorphism. Therefore, I think the diffeo should be defined in the metric and not in the manifold, which will lead to different logics between `PullbackMetric` and `PullbackDiffeoMetric`.

Reply:
Regarding the `init`, all the metrics have an init. Most of the times it is the same as `RiemannianMetric`, so we don't explictly write it. Other times we have to perform some more operations besides the ones done by `RiemannianMetric` and `Connection`. That's the case here.